index.rst

```diff
@@ -470,93 +520,84 @@ capability token may be passed in the URL with the HTTP Basic Auth scheme.
+impersonation for finer-grained authorization.
 
-In Phase 2, the token issuer will be presented with an identity token by a service, either the
-portal or some third-party application or library, and MUST issue a refresh token. The refresh token
-can be presented at any time to the token issuer for a capability token.
+In both approaches, services in the LSP API aspect may also be responsible for inspecting
+the token for groups of interest or capabilities, but the token could be assumed to be validated.
 
-.. todo:: `Service provided by data publisher 
-          Uses identity/refresh token to issue refresh/access token 
-          For our purposes, has a fixed list of scopes plus scopes derived from LDAP groups (no
-          actual separate policy database needed) 
-          Limits scope to what client and user request/allow`
+Token Proxy
+-----------
 
-Token Authorizer
-----------------
+The Token Proxy is a single gateway to which can handle OAuth2 authentication flows, and integrate
+with the `token authorization <#token-authorization>`__ and `token issuer <#token-issuer>`__
+components. Critically, through the token issuer, the token proxy implements transparent token
+reissuance for downstream services.
 
-All LSP services are responsible for validating tokens. For Phase 1, the portal and notebook are
-responsible for inspecting the token for any groups of interest, or delegating to a service, to
-control access to the service. The LSP API aspect is responsible for verifying the token received,
-as well as also inspecting the token for any groups of interest. Services in the LSP API aspect are
-also responsible for impersonation for the underlying systems.
+The reissued token MAY alter the values of the following ``iss``, ``exp``, and ``iat`` claims. All
+other claims MUST be included in the reissued token, unmodified. Additional claims may also be
+included.
 
-In Phase 2, services in the LSP API aspect will rely on capabilities in the ``scope`` claim of the
-capability token to limit access to the requisite service. It will then rely on impersonation for
-finer-grained authorization.
+When reissuing tokens, the token proxy MUST make those headers available to the downstream
+services via HTTP Headers.
 
-Token Proxy
------------
+* ``X-Auth-Request-Token: [token]``
+* ``Authorization: Bearer [token]``
 
-The LSP API Aspect MUST be able to make requests to other services. This requires relaying the
-appropriate tokens to the services. In order to satisfy a `token acceptance
-guarantee <#token-acceptance-guarantee>`__, in the context of asynchronous and long-running requests,
-the LSP API Aspect MUST obtain, either through self-issuance or a request to the `token
-issuer <#token-issuer>`__, a new token with a bounded lifetime which can be honored by the other LSP
-API aspect services.
+The token proxy, or it's conceptual deployment, must be flexible enough to set those headers to a
```

Comment:
Remove "or it's conceptual deployment".

Reply:
I think I should rewrite this to just omit the token renaming flexibility. The token proxy itself doesn't actually set the headers to different values before sending it to the backend - the downstream service (aka notebook) does that via the nginx ingress rules. The notebook, for example, does that because Jupyterlab wants to send headers of the form `Authorization: token [token]` and our `Authorization` tokens were stomping on that.